app/controllers/admin/schedules_controller.rb

```diff
@@ -31,7 +31,16 @@ def index
 
     def collection
       return Schedule.where("1=0") unless json_request?
-      permissions.visible_schedules
+      if params[:enterprise_id]
+        filter_schedules_by_enterprise_id(permissions.visible_schedules, params[:enterprise_id])
+      else
+        permissions.visible_schedules
+      end
+    end
+
+    # Filter schedules by OCs with a given coordinator id
+    def filter_schedules_by_enterprise_id(schedules, enterprise_id)
+      schedules.where("order_cycles.coordinator_id = ?", enterprise_id.to_i)
```

Comment:
is this better @sauloperez ?

Reply:
I find that much better. You are relying on the permissions class joining with order cycles but that's okay. You could have the same join here and Rails would filter out the duplicate join. And you can use the same hash syntax:
```rb
schedules.joins(:order_cycles).where(order_cycles: { coordinator_id: enterprise_id.to_i })
```

We don't need the distinct select because coordinators are uniq on each order cycle.

Reply:
ah, nice, I do that! new commit.